common/config/src/main/java/io/helidon/common/config/Config.java

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2018, 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.config;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <h2>Configuration</h2>
+ * Immutable tree-structured configuration.
+ *
+ * See {@link ConfigValue}.
+ */
+public interface Config {
+
+    /**
+     * Returns the fully-qualified key of the {@code Config} node.
+     * <p>
+     * The fully-qualified key is a sequence of tokens derived from the name of
+     * each node along the path from the config root to the current node. Tokens
+     * are separated by {@code .} (the dot character). See {@link #name()} for
+     * more information on the format of each token.
+     *
+     * @return current config node key
+     * @see #name()
+     */
+    BaseKey key();
```

Comment:
BaseKey? This should return `Key`. Our API should be `Key`, `BaseKey` I would expect to be an abstract class I can extend, not API